hyperion/serializers.py

```diff
@@ -168,19 +167,19 @@ def create(self, validated_data):
         visible_to_data = validated_data.pop("visible_to", [])
         post = Post.objects.create(author=user.profile, **validated_data)
         visible_to = [
-            User.objects.get(username=username).profile.pk for username in visible_to_data
+            User.objects.get(username=username).profile.url for username in visible_to_data
```

Comment:
This won't work for foreign users, I think?

Foreign users only have an entry with in the `UserProfile` table, but nothing in the `User` table.

cc @haotianzhu

Reply:
fixed